File: Source/Fantasy.Base/System.Reflection/MemberInfoExtension.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace System.Reflection
{
    public static class MemberInfoExtension
    {
        public static T[] GetCustomAttributes<T>(this MemberInfo mi, bool inherit)
        {
            return mi.GetCustomAttributes(typeof(T), inherit).Cast<T>().ToArray();
        }
    }
}

File: branches/Stheno/AutoConverted/Fantasy.Base/System.Reflection/MemberInfoExtension.java
package System.Reflection;

public final class MemberInfoExtension
{
//C# TO JAVA CONVERTER TODO TASK: Extension methods are not available in Java:
//ORIGINAL LINE: public static T[] GetCustomAttributes<T>(this MemberInfo mi, bool inherit)
	public static <T> T[] GetCustomAttributes(MemberInfo mi, boolean inherit)
	{
		return mi.GetCustomAttributes(T.class, inherit).<T>Cast().toArray();
	}
}